### PR TITLE
scheduler: fix sort hotpeer in hot scheduler

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

fix sort hotpeer in hot scheduler

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]
I think this section is not covered by our tests

**Comment by lhy1024** [comment]
> This section is not covered by our tests :(
I will add the test later

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
@lhy1024  statics failed.

File: server/schedulers/hot_test.go [hunk 1]
```diff
@@ -1267,6 +1267,64 @@ func (s *testHotCacheSuite) TestCheckRegionFlowWithDifferentThreshold(c *C) {
 	}
 }
 
+func (s *testHotCacheSuite) TestSortHotPeer(c *C) {
+	ctx, _ := context.WithCancel(context.Background())
```
Comment (rleungx) [inline server/schedulers/hot_test.go:1271]: defer cancel()?

File: server/schedulers/hot_test.go [hunk 2]
```diff
@@ -1267,6 +1267,64 @@ func (s *testHotCacheSuite) TestCheckRegionFlowWithDifferentThreshold(c *C) {
...
+	opt := config.NewTestOptions()
+	tc := mockcluster.NewCluster(ctx, opt)
+	tc.SetMaxReplicas(3)
+	tc.SetLocationLabels([]string{"zone", "host"})
```
Comment (rleungx) [inline server/schedulers/hot_test.go:1275]: Why?

File: server/schedulers/hot_test.go [hunk 3]
```diff
@@ -1267,6 +1267,64 @@ func (s *testHotCacheSuite) TestCheckRegionFlowWithDifferentThreshold(c *C) {
...
+			},
+		},
+	}
+
+	u := leaderSolver.sortHotPeers(hotPeers, 1)
+	checkSortResult(c, []uint64{1}, u)
+
+	u = leaderSolver.sortHotPeers(hotPeers, 2)
+	checkSortResult(c, []uint64{1, 2}, u)
+
```
Comment (rleungx) [inline server/schedulers/hot_test.go:1311]: remove the line

**Review by rleungx — APPROVED** [review]

**Pull request merged**